File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stack = []
        n = len(part)
        for i in range(len(s)):
            stack.append(s[i])
            sub = ''.join(stack[-n:])
            if sub == part:
                del stack[-n:]
        return ''.join(stack)

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function (s, part) {
    let stack = []
    let n = part.length
    for (let i = 0; i < s.length; i++) {
        stack.push(s[i])
        const sub = stack.slice(-n).join("")
        if (sub == part) {
            stack.splice(stack.length - n, n)
        }
    }
    return stack.join("")
};